### ISSUE TITLE
Notification of Loading/Busy: Question about the initial wait to load a new page

### ISSUE THREAD

**Issue by kimamula** [w3c/wcag#4789]
Hi,

I have a question regarding this document: https://www.w3.org/WAI/GL/wiki/Notification_of_Loading/Busy

It says that: "This is not about the initial wait to load a new page. Those are announced to screen readers."

Does this mean that screen readers are responsible for announcing the loading state during the initial page load, and web pages shouldn't need to implement programmatic notifications for that?

We have a web page that takes time to load (especially for mobile devices). When a user visits this page, they first see a loading spinner that is included in the HTML returned from the server for like several seconds, and then see the actual page after the page completes loading the required JS and CSS resources. The accessibility team at my organization pointed out that it is an accessibility violation that screen readers do not announce the loading state when the loading spinner is displayed, referring to https://www.w3.org/WAI/GL/wiki/Notification_of_Loading/Busy.

I investigated this issue with VoiceOver and found that:
- `aria-live` doesn't work until the window's `load` event (i.e., VoiceOver doesn't announce anything when the content of a live region is updated before the `load` event). Since the majority of the time the loading spinner is displayed is before the `load` event, `aria-live` doesn't help here.
- Focusing programmatically on a visually hidden text "Loading" immediately after the HTML load lets VoiceOver announce it. However, I'm not confident that this is a proper solution because I can't find anyone mentioning such a solution on the internet. (Also, I'm unsure if this works for all screen readers.)

Could you please guide me on how web pages are supposed to be implemented in this case?

Thank you!

**Comment by TestPartners** [comment]
Your interpretation is correct and your accessibility team are wrong. Under the SC Text heading, the document says, “After initial page load, programmatic notification is provided for each visual indicator that content is loading or the page is busy”.

The document only applies to changes that occur without a page reload, such as AJAX or single page applications.

**Comment by patrickhlauke** [comment]
> Your interpretation is correct and your accessibility team are wrong. Under the SC Text heading, the document says, “After initial page load, programmatic notification is provided for each visual indicator that content is loading or the page is busy”.

based on the OP's description, the "initial page load" is when the page is first shown with the spinner. it then makes dynamic calls to actually load all the CSS/JS? (at least i hope so, as a page shouldn't need "several seconds" to actually load CSS/JS unless it's seriously unoptimised)

if this spinner is only a generic animated indicator, then it needs to at least have an appropriate accessible name/text alternative (not related to 4.1.3 Status Messages, but more generally 1.1.1 Non-text Content).

if the spinner does actually show/convey information, such as changing to indicate how much of the loading process completed, then this *will* need to be dynamically announced.
